youtube_dl/extractor/ndtv.py

```diff
@@ -22,21 +26,22 @@ class NDTVIE(InfoExtractor):
             'upload_date': '20131208',
             'duration': 1327,
             'thumbnail': r're:https?://.*\.jpg',
-        },
+        }
     }
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
         webpage = self._download_webpage(url, video_id)
 
-        title = remove_end(self._og_search_title(webpage), ' - NDTV')
+        title = compat_urllib_parse_unquote_plus(
+            self._search_regex(r"__title\s*=\s*'([^']+)'", webpage, 'title'))
```

Comment:
Again: **do not remove the old code or change the order**.

Reply:
But the "__title" regex would give more accurate results in this context.

Reply:
No, prove. At best it produces the same string.

Reply:
With url http://profit.ndtv.com/videos/news/video-indian-economy-on-very-solid-track-international-monetary-fund-chief-470040:
* `_og_search_title` output:
```
$ python -m youtube_dl 'http://profit.ndtv.com/videos/news/video-indian-economy-on-very-solid-track-international-monetary-fund-chief-470040'
[NDTV] 470040: Downloading webpage
[download] Destination: Indian Economy On 'Very Solid Track' - International Monetary Fund Chief Video-470040.mp4
[download] 100% of 6.46MiB in 00:01
```

* `_search_regex` output:
```
$ python -m youtube_dl 'http://profit.ndtv.com/videos/news/video-indian-economy-on-very-solid-track-international-monetary-fund-chief-470040'
[NDTV] 470040: Downloading webpage
[download] Destination: Indian Economy On 'Very Solid Track' - International Monetary Fund Chief-470040.mp4
[download] 100% of 6.46MiB in 00:01
```

Note the extra "Video" in the first case.

**Another example**
* `_og_search_title` output:
```
$ python -m youtube_dl 'https://khabar.ndtv.com/video/show/prime-time/prime-time-ill-system-and-poor-education-468818'
[NDTV] 468818: Downloading webpage
[download] Destination: प्राइम टाइम - सिस्टम बीमार, स्कूल बदहाल वीडियो - हिन्दी न्यूज़ वीडियो एनडीटीवी ख़बर-468818.mp4
[download] 100% of 115.69MiB in 00:27
```
* `_search_regex` output:
```
$ python -m youtube_dl 'https://khabar.ndtv.com/video/show/prime-time/prime-time-ill-system-and-poor-education-468818'
[NDTV] 468818: Downloading webpage
[download] Destination: प्राइम टाइम - सिस्टम बीमार, स्कूल बदहाल-468818.mp4
[download] 100% of 115.69MiB in 00:25
```

Again, the string "वीडियो - हिन्दी न्यूज़ वीडियो एनडीटीवी ख़बर" (which translates to 'Video - Hindi news video NDTV khabar) is extra in the first case.

The existing code is manually removing the trailing " - NDTV".  The new code automatically takes care of that.

Reply:
Then add a comment that clearly describes the rationale and a test for this scenario with unambiguous comment as well. Current title code should become a fallback and must be reachable.

Reply:
I think the second url would be a better option as it would also test the `urllib_parse.unquote_plus` part.

`_search_regex` will return `None` if no match is found (according to common.py:800).  In that case the `or` operator would choose the fallback option.  If you have something else in my mind please suggest it.

Reply:
800 won't be reached due to `fatal == True`.

Reply:
Yes, should've mentioned that.  I added `fatal=False`.

I guess similar fallback will have to be added for `__filename`.  It fails on the new formats you mentioned below.

Reply:
`fatal=False` will cause wrong duplicate warnings in case of failures. `default=None` should be used instead.